wgsl/index.bs

```diff
@@ -513,7 +514,205 @@ When unclear from context, this specification indicates
+    * If such a filter exists, apply it to *D*, updating *D*'s [=diagnostic/severity=].
+    * Otherwise *D* remains unchanged.
+2. Discard diagnostics that have severity [=severity/off=].
+3. If at least one remaining diagnostic *DI* has severity [=severity/info=], then:
+    * Other [=severity/info=] diagnostics with same triggering rule *may* be discarded, leaving only the original diagnostic *DI*.
+4. If at least one remaining diagnostic *DW* has severity [=severity/warning=], then:
+    * Other [=severity/info=] or [=severity/warning=] diagnostics with same triggering rule *may* be discarded, leaving only the original diagnostic *DW*.
+5. If at least one remaining diagnostic has [=severity/error=] severity, then:
+    * Other diagnostics *may* be discarded, including other diagnostics with [=severity/error=] severity.
+    * A [=program error=] is generated.
+         * The error is a [=shader-creation error=] if the diagnostic was triggered at [=shader module creation=] time.
+         * The error is a [=pipeline-creation error=] if the diagnostic was triggered at [=pipeline creation=] time.
+6. If processing during [=shader module creation=] time, the remaining diagnostics populate the
+    {{GPUCompilationInfo/messages}} member of the WebGPU {{GPUCompilationInfo}} object.
+7. If processing during [=pipeline creation=], the [=severity/error=] diagnostics are reported
+    in the nearest enclosing WebGPU [=GPU error scope=].
+    Issue: TODO: Adjust for [#3682](https://github.com/gpuweb/gpuweb/issues/3682).
+
+Note: The rules allow an implementation to stop processing a WGSL program as soon as an error is detected.
+Additionally, an analysis for a particular kind of warning can stop on the first warning,
+and an analysis for a particular kind of info diagnostic can stop on the first occurrence.
+WGSL does not specify the order to perform different kinds of analyses, or an ordering within a single analysis.
+Therefore, for the same WGSL program, different implementations may report different instances of diagnostics with the same severity.
+
+### Filterable Triggering Rules ### {#filterable-triggering-rules}
+
+Most [=diagnostics=] are unconditionally reported to the WebGPU application.
+Some kinds of diagnostics can be [[#diagnostic-filtering|filtered]], in part by naming their [=diagnostic/triggering rule=].
+The following table lists the standard set of triggering rules that can be filtered.
+
+<table class='data'>
+  <caption>Filterable diagnostic triggering rules</caption>
+  <thead>
+    <tr><th>Filterable Triggering Rule<th>Default Severity<th>Triggering Location<th>Description
+  </thead>
+
+  <tr>
+    <td><dfn noexport dfn-for="trigger">derivative_uniformity</dfn>
+    <td>[=severity/error=]
+    <td>The [=call site=] for any of the
+        [=builtin functions that compute a derivative|builtin function that computes a derivative=].
```

Comment:
Nit: this reads a little strange in the rendered version:
"for any of the builtin function that computes a derivative"
Suggest removing "of the".